web/src/pages/onboarding/components/OnboardingBottomNav.tsx

```diff
@@ -44,7 +46,11 @@ export const OnboardingBottomNav: React.FC<Props> = ({
       className="flex w-full justify-between mb-4 h-8 self-end select-none"
     >
       <div className="flex self-center">
-        <BackButton onClick={removeOnboardingStep} disabled={backDisabled} data-cy="prev-onboarding-step" />
+        <BackButton
+          onClick={isCompleted ? removeTwoOnboardingStep : removeOnboardingStep}
+          disabled={backDisabled}
+          data-cy="prev-onboarding-step"
+        />
```

Comment:
Please add inline comments with the rationale for this, will not be self-evident to future generations.

Reply:
Added.

Reply:
I'd put the comment right next to the part that is likely to cause confusion (the one-step two-step conditional).

Reply:
Done